File: Ruby/problem030.rb
#! /usr/bin/env ruby
=begin
Problem 30 - Project Euler
http://projecteuler.net/index.php?section=problems&id=30
=end
puts (100..(9**5*6)).select {|i|
    i == i.to_s.chars.inject(0) {|sum,c| sum + c.to_i ** 5 }
}.sum

File: Kotlin/problem030.kt
// Problem 30 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=30
fun Int.nth_power(n: Int): Int {
    var x = 1
    repeat(n) { x *= this }
    return x
}

fun main() {
    val sum = (2..354294)
            .map { x ->
                Pair(x, x.toString()
                        .map { c -> (c.code - 48).nth_power(5) }
                        .sum())
            }
            .filter { it.first == it.second }
            .map { it.first }
            .sum()
    println(sum)
}
